.github/scripts/binaryCompatibility.py

```diff
@@ -0,0 +1,76 @@
+#  Copyright Amazon.com Inc. or its affiliates.
+#  SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import json
+import os
+import xml.etree.ElementTree as ET
+
+from agithub.GitHub import GitHub
+
+
+def findall_recursive(node, element):
+    for item in node.findall(element):
+        yield item
+    for item in list(node):
+        yield from findall_recursive(item, element)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--input', type=argparse.FileType('r'), help='Input file to parse')
+    parser.add_argument('--token', type=str, help='GitHub token')
+    args = parser.parse_args()
+
+    file = args.input
+
+    my_body_dedupe = "Produced by binaryCompatability.py"
+    body = f"Binary incompatibility detected for commit {os.getenv('GITHUB_SHA')}.\n" \
+           f" See the uploaded artifacts from the action for details. You must bump the version number.\n\n"
+
+    incompatible = False
+    tree = ET.parse(file)
+    for clazz in tree.getroot().findall("./classes/class"):
+        any_incompatible = False
+        body += f"{clazz.attrib['fullyQualifiedName']} is "
+        if clazz.attrib['binaryCompatible'] == "false":
+            body += "binary incompatible"
+            incompatible = True
+            any_incompatible = True
+        if clazz.attrib['sourceCompatible'] == "false":
+            if any_incompatible:
+                body += " and is "
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are concatenating a string inside a loop. Concatenating immutable sequences will always results in a new object. This means that building up a sequence by repeated concatenation will have a quadratic runtime cost in the total sequence length. For efficiency and to get a linear runtime cost, consider appending values to a list and concatenating that list using [str.join()](https://docs.python.org/3/library/stdtypes.html#str.join) or [bytes.join()](https://docs.python.org/3/library/stdtypes.html#bytes.join) method.

[Learn more](https://docs.python.org/3/faq/programming.html#what-is-the-most-efficient-way-to-concatenate-many-strings-together)

Similar issue at line number 50.